### PR TITLE
Don't auto-approve versions pending rejection

### PR DESCRIPTION
**Pull request by diox**
Fixes: mozilla/addons#14936

### Context

Every add-on version that is eligible for auto-approval goes through `AutoApprovalSummary.create_summary_for_version()` to create an `AutoApprovalSummary` instance, and then through `<instance>.calculate_verdict()`.

`create_summary_for_version()` calls all `check_<field>` methods from `AutoApprovalSummary.auto_approval_verdict_fields`, setting the relevant field on the instance (so that we keep a record of why something wasn't auto-approved). `calculate_verdict()` checks that all those fields are `False` to decided whether or not the version should be auto-approved.
 
This PR adds a new field to `AutoApprovalSummary` , `is_pending_rejection`, as well as the corresponding method, `check_is_pending_rejection()`. That methods return `False` for versions pending rejection, preventing their auto-approval, fixing the issue.

### Testing
https://github.com/mozilla/addons/issues/14936#issuecomment-2304725377

### CODE REVIEW

File: src/olympia/reviewers/tests/test_models.py [hunk 1]
```diff
@@ -1315,6 +1315,22 @@ def test_check_is_locked(self):
...
+        assert AutoApprovalSummary.check_is_pending_rejection(self.version) is False
+
+        flags.update(
+            pending_rejection=datetime.now() + timedelta(hours=1),
+            pending_rejection_by=user_factory(),
+            pending_content_rejection=False,
+        )
+        assert AutoApprovalSummary.check_is_pending_rejection(self.version) is True
+
+        flags.update(pending_content_rejection=True)
```
[the remark this answers is not in the document]
Comment (diox) [reply src/olympia/reviewers/tests/test_models.py:1331]: I'm checking that changing the `pending_content_rejection` field has no effect - `check_is_pending_rejection()` should continue to return `True` in that case. We could split it up but I don't think it matters since that's the only change - this style is consistent with the other tests in that class

[the remark this answers is not in the document]
Comment (KevinMind) [reply src/olympia/reviewers/tests/test_models.py:1331]: Okay so the test is actually that updating content rejection to true should NOT have an effect. Okay. It wasn't super clear in the test but it makes sense now that you mention. Maybe worth a comment for future clarity but up to you.

**Comment by KevinMind** [comment]
It's not totally clear to me how this PR satisfies the ticket's requirements. I see we added the model method, with tests, but where is it used currently? Can you add some context how exactly this PR fixes the issue?

**Comment by diox** [comment]
@KevinMind sorry, should have done that first, I know that code inside out but it can be a bit hard to follow if you're not familiar with it. I've added some context, let me know if that's enough.

**Comment by KevinMind** [comment]
Tried verifying according to the STR.. not quite working. Would like some feedback before approving.

**Review by KevinMind — APPROVED** [review]
Verified the issue was unrelated to this PR but to removing local files in ./storage.
